### PR TITLE
feat: sample message too large to s3

### PR DESCRIPTION
**Pull request by pauldambra**
We get `MessageTooLarge` error from Kafka, we know this means that the event content is too large for kafka, and based on how we process messages we can be pretty sure this is caused by full snapshots (which aren't chunk-up-able)

This adds a new sample rate config to dump a selection of these messages to S3 so we can analyse them

* the sample rate must be over 0
* it must be <= 0.001 (we want to be very sure that we don't overwhelm capture)
* we wrap the s3 upload in a try/except block to try (pun intended) not to break capture unexpectedly

I'm anticipating manually setting the environment variable and monitoring the bucket until we have >10 and <100 samples and then turning it off so we can validate the contents

once we've gathered enough samples to take some remedial steps in the SDK or during ingestion, we'll remove this code

### CODE REVIEW

File: posthog/api/capture.py [hunk 1]
```diff
@@ -677,6 +680,23 @@ def replace_with_warning(event: dict[str, Any]) -> dict[str, Any] | None:
...
 
+def sample_replay_data_to_object_storage(event: dict[str, Any], random_number: float) -> None:
+    """
+    the random number is passed in to make testing easier
+    both the random number and the sample rate must be between 0 and 0.01
+    if the random number is less than the sample_rate then we write the event to S3
+    """
+    try:
+        sample_rate = settings.REPLAY_MESSAGE_TOO_LARGE_SAMPLE_RATE
+        if 0 < random_number < sample_rate <= 0.01:
```
Comment (pauldambra) [inline posthog/api/capture.py:691]: i'm consistently amazed by how nice some parts of python syntax are

File: posthog/api/capture.py [hunk 2]
```diff
@@ -677,6 +680,23 @@ def replace_with_warning(event: dict[str, Any]) -> dict[str, Any] | None:
         return None
 
 
+def sample_replay_data_to_object_storage(event: dict[str, Any], random_number: float) -> None:
+    """
+    the random number is passed in to make testing easier
+    both the random number and the sample rate must be between 0 and 0.01
```
Comment (marandaneto) [inline posthog/api/capture.py:686]: Usually, sample rate is more human-readable when it's between 0...1, so 0.1 is 10% and 1 is 100%, It's not a blocker, just the standard I've seen across different tools.

**Review by marandaneto — APPROVED** [review]
Good idea.

File: posthog/api/capture.py (hunk 2 above)
Comment (pauldambra) [reply posthog/api/capture.py:686]: Ah, yep.. so this is still between 0 and 1 but clamped to 0.01
I don't want to upload 100s of messages a minute if someone misconfigures this value

**Pull request merged**